Docs/prewritten policies (#792)
hashicorp/terraform-docs-common: website/docs/cloud-docs/policy-enforcement/define-policies/prewritten-sentinel.mdx

1. Complete the instructions for [connecting the policies to your workspace](#connect-policies-to-your-workspace).

### Create a fork of the policy libraries 

Create a fork of the repository containing the policies you want to implement. Refer to the [GitHub documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo) for instructions on how to create a fork. 

HashiCorp Sentinel policy libraries include a `sentinel.hcl` file. You can modify the file to customize your policy set. Refer to [Sentinel Policy Set VCS Repositories](/terraform/cloud-docs/policy-enforcement/manage-policy-sets/sentinel-vcs) for additional information.

After forking the repository, complete the instructions for [connecting the policies to your workspace](#connect-policies-to-your-workspace).

### Add the Terraform module to your configuration 
This method enables you to connect the policies to workspaces in the Terraform configuration file. As a result, you can skip the instructions described in [Connect policies to your workspaces](#connect-policies-to-your-workspaces). 
	 

> 1. Go to the [module in the Terraform registry](https://registry.terraform.io/modules/hashicorp-modules/aws-cis/sentinel/latest) and copy the code generated in the **Provision Instructions** tile.  

Author: ```suggestion
1. Go to the [module in the Terraform registry](https://registry.terraform.io/modules/hashicorp/CIS-Policy-Set/AWS/latest) and copy the code generated in the **Provision Instructions** tile.  
```

## Overview

Pre-written Sentinel policy libraries streamline your compliance processes and enhance security across your infrastructure. HashiCorp's ready-to-use policies can help you enforce best practices and security standards across your AWS environment.

Refer to the following resources for details about working with pre-written policies and information about the Sentinel language and framework: 

- [Sentinel documentation](/sentinel/docs) 
- README documentation included with each of the policy libraries.

Complete the following steps to implement pre-written Sentinel policies in your workspaces:

1. Obtain the policies you want to implement: You download policies directly into your repository or create a fork of the HashiCorp repositories. Alternatively, you can add the Terraform module to your configuration, which acquires the policies and connects them to your workspaces in a single step.
1. Connect policies to your workspace. If you downloaded policies or forked one or more repositories, you must manually connect the repository containing the policies to your HCP Terraform or Terraform Enterprise workspaces.       

## Requirements

> You must use one of the following Terraform applications:

Author: I think direct is clearer.

- HCP Terraform
- Terraform Enterprise v202312-1 or newer

### Permissions

Your HCP Terraform or Terraform Enterprise user account must either be a member of the owners team or have the **Manage Policies** organization-level permissions enabled to create new policy sets and policies. Refer to the following topics for additional information:

- [Organization owners](/terraform/cloud-docs/users-teams-organizations/permissions#organization-owners)
- [Manage policies](/terraform/cloud-docs/users-teams-organizations/permissions#manage-policies)

### Version control system

> You must have a GitHub account connected to HCP Terraform or Terraform Enterprise to manually connect policy sets to your workspaces. Refer to [Connecting VCS Providers](/terraform/cloud-docs/vcs) for instructions.

Reviewer: Is this GitHub only or will it work with any VCS?

Author: GitHub only. I'll add it to the requirements.

## Obtain policies

You can use the policy libraries created and maintained by HashiCorp. The libraries are stored in the following GitHub repositories:

- [policy-library-cis-aws-efs-terraform](https://github.com/hashicorp/policy-library-cis-aws-efs-terraform)
- [policy-library-cis-aws-rds-terraform](https://github.com/hashicorp/policy-library-cis-aws-rds-terraform)
- [policy-library-cis-aws-vpc-terraform](https://github.com/hashicorp/policy-library-cis-aws-vpc-terraform)
- [policy-library-cis-aws-iam-terraform](https://github.com/hashicorp/policy-library-cis-aws-iam-terraform)
- [policy-library-cis-aws-s3-terraform](https://github.com/hashicorp/policy-library-cis-aws-s3-terraform)
- [policv-library-cis-aws-cloudtrail-terraform](https://github.com/hashicorp/policv-library-cis-aws-cloudtrail-terraform)
- [policy-library-cis-aws-kms-terraform](https://github.com/hashicorp/policy-library-cis-aws-kms-terraform)
- [policy-library-cis-aws-ec2-terraform](https://github.com/hashicorp/policy-library-cis-aws-ec2-terraform)

> Use one of the following methods to obtain pre-written policies:

Reviewer: Should we give users some context about when to use which way? It might be confusing to present them without pros and cons, especially since using the module seems so much easier than the others?

> - Download policies from the registry
> - Fork the HashiCorp policy GitHub repository
> - Add the Terraform module to your configuration 

Author: ```suggestion
- **Download policies from the registry**:  Use this method if you want to assemble custom policy sets without customizing policies.  
- **Fork the HashiCorp policy GitHub repository**: Use this method if you intend to customize the policies. 
- **Add the Terraform module to your configuration**: Use this method to implement specific versions of the policies as-is. This method also connects the policies to workspaces in the Terraform configuration file instead of connecting them as a separate step. 
```

1. Complete the instructions for [connecting the policies to your workspace](#connect-policies-to-your-workspace).

### Create a fork of the policy libraries 

Create a fork of the repository containing the policies you want to implement. Refer to the [GitHub documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo) for instructions on how to create a fork. 

HashiCorp Sentinel policy libraries include a `sentinel.hcl` file. You can modify the file to customize your policy set. Refer to [Sentinel Policy Set VCS Repositories](/terraform/cloud-docs/policy-enforcement/manage-policy-sets/sentinel-vcs) for additional information.

After forking the repository, complete the instructions for [connecting the policies to your workspace](#connect-policies-to-your-workspace).

### Add the Terraform module to your configuration 
This method enables you to connect the policies to workspaces in the Terraform configuration file. As a result, you can skip the instructions described in [Connect policies to your workspaces](#connect-policies-to-your-workspaces). 
	 

> 1. Go to the [module in the Terraform registry](https://registry.terraform.io/modules/hashicorp-modules/aws-cis/sentinel/latest) and copy the code generated in the **Provision Instructions** tile.  

Author: I tracked down the correct link.

1. Add the `module` block to your Terraform configuration and define the following arguments: 
   - `source`: Specify the path to the module you downloaded.
   - `tfe_organization`: Specify the name of your organization on Terraform Enterprise or HCP Terraform.
   - `policy_set_workspace_names`: Specify a list of workspace names that you want to apply the policies to. 

   The following example configuration applies invokes the module for `target_workspace_1`:

   ```hcl
   module "cis_v1-2-0_policies" {
      source = "../prewritten-policy"
      name                             = "cis-1-2-0"
      tfe_organization                 = "<your-tfe-org>"
      policy_set_workspace_names       = ["target_workspace_1"]
   }
   ```

1. Run `terraform plan` to view the plan.
1. Run `terraform apply` to apply the changes.

> After successfully running the `terraform apply` command, Terraform evaluates Sentinel policies for each run of the workspaces you specified.

Reviewer: Are we referring to the apply command the user just ran, or apply commands in the workspaces they just added the policies to?

> 1. Run `terraform apply` to apply the changes.
> 
> After successfully running the `terraform apply` command, Terraform evaluates Sentinel policies for each run of the workspaces you specified.

Author: ```suggestion
1. Run `terraform apply` to apply the changes. After running the command, Terraform will evaluate Sentinel policies for each following run of the workspaces you specified.
```

## Connect policies to your workspace

> Skip this step if you [added the Terraform module](#add-the-terraform-module-to-your-configuration) to your configuration. When you use the module, you programmatically connect policies to workspaces in the HCL. 

Reviewer 2: ```suggestion
Skip this step if you [added the Terraform module](#add-the-terraform-module-to-your-configuration) to your configuration. When you use the module, setting the `policy_set_workspace_names` argument causes Terraform to connect the policies to your specified HCP Terraform workspaces. 
```

Reviewer 2: Or something, just breadcrumbs so the user can refer back to the previous step to understand the connection here.

1. Log into your organization and click **Settings** in the sidebar. 
1. Click **Policy Sets** and click **Connect a new policy set**.
1. Click the **Version control provider (VCS)** tile.
1. Enable the **Sentinel** option as the policy framework.

> 1. Specify a name and description for the set.

Reviewer 2: ```suggestion
1. Specify a name and optionally a description for the set.
```

Author: Let's be prescriptive

1. Configure any additional options for the policy set and click **Next**.
1. Choose the GitHub connection type, then choose the repository you created in [Set up a repository for the policies](#set-up-a-repository-for-the-policies). 

> 1. If you stored the `sentinel.hcl` policy set file in folder, specify the path to the file in the **Policies path** field. The default is the root directory.

Reviewer 2: ```suggestion
1. If the selected repository includes the `sentinel.hcl` policy set file in subfolder, specify the path to the file in the **Policies path** field. The default is the repository's root directory.
```

Author: ```suggestion
1. If the `sentinel.hcl` policy set file is stored in a subfolder, specify the path to the file in the **Policies path** field. The default is the root directory.
```

> 1. If you want to apply updated policies sets to the workspace from a specific branch, specify the name in the **VCS branch** field.  The default is the default branch configured for the repository.

Reviewer 2: ```suggestion
1. If you want to apply updated policy sets to the workspace from a specific branch, specify the name in the **VCS branch** field. The default is the default branch configured for the repository.
```

1. Click **Next** and specify any additional parameters you want to pass to the Sentinel runtime and click **Connect policy set** to finish applying the policies to the workspace. 

> Run a plan in the workspace to trigger the connected policies. Refer to [Start a Terraform run](/terraform/cloud-docs/run/remote-operations#starting-runs) for additional information.

Reviewer 2: It might be nice to explain why and when you would do this, eg: "Some policies include configuration parameters to customize their behavior. For example, the CIS repository used in the previous example includes parameters to..."

## Next steps

> - Group your policies into sets and apply them to your workspaces. Refer to [Create policy sets](/terraform/cloud-docs//policy-enforcement/create-sets) for additional information.
> View results and address Terraform runs that do not comply with your policies. Refer to - [View results](/terraform/cloud-docs/policy-enforcement/view-results) for additional information.
> - You can also view Sentinel policy results in JSON format. Refer to [View Sentinel JSON results](/terraform/cloud-docs/policy-enforcement/view-json-results) for additional information. 

Reviewer 2: ```suggestion
- Group your policies into sets and apply them to your workspaces. Refer to [Create policy sets](/terraform/cloud-docs//policy-enforcement/create-sets) for additional information.
- View results and address Terraform runs that do not comply with your policies. Refer to [View results](/terraform/cloud-docs/policy-enforcement/view-results) for additional information.
- You can also view Sentinel policy results in JSON format. Refer to [View Sentinel JSON results](/terraform/cloud-docs/policy-enforcement/view-json-results) for additional information. 
```
